app/validators/declaration_date_validator.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class DeclarationDateValidator < ActiveModel::Validator
```

Comment:
a custom validator, to make separate the validation of a component of the request request (i.e declaration_date) so that it does not clutter the service class.